### PR TITLE
Extra conformance test for empty case

### PR DESCRIPTION
**Pull request by kwxm**
This just adds an extra conformance test to check that `(lam x (case x))` succeeds.  This is an edge case that was unclear in the specification until recently (see #7141) and which we weren't previously checking.

### CODE REVIEW

File: plutus-conformance/test-cases/uplc/evaluation/term/case/case-09/case-09.uplc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- It's legal to have a case term with no branches,
+-- but applying it causes an error.
```
Comment (SeungheonOh) [inline plutus-conformance/test-cases/uplc/evaluation/term/case/case-09/case-09.uplc]: Probably I'm dumb but this took me a while to process. Maybe we can use a clearer wording: "but casing anything on no-branch case term causes an error".

Comment (kwxm) [reply plutus-conformance/test-cases/uplc/evaluation/term/case/case-09/case-09.uplc]: OK, I've tried to clarify that.

**Review by SeungheonOh — APPROVED** [review]

**Comment by kwxm** [comment]
I'm going to bypass CI here too (too many failures).
